crypto/o_fips.c

```diff
@@ -1,24 +1,49 @@
 /*
- * Copyright 2011-2016 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2011-2020 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
  */
 
+#include "crypto/evp.h"
 #include "internal/cryptlib.h"
 
+/* Check that the global properties has the value "fips=yes" */
+int FIPS_mode_with_ctx(OPENSSL_CTX *libctx)
```

Comment:
We already have SSL_CTX_new_with_libctx, so probably we should go with `FIPS_mode_with_libctx`